src/async_fuse/memfs/s3_metadata.rs

```diff
@@ -370,114 +393,76 @@ impl<S: S3BackEnd + Sync + Send + 'static> MetaData for S3MetaData<S> {
-                            .open_child_file(
-                                child_name,
-                                child_attr,
-                                oflags,
-                                Arc::<GlobalCache>::clone(&self.data_cache),
-                            )
-                            .await
-                            .context(format!(
-                            "lookup_helper() failed to open child file name={child_name:?} with flags={oflags:?} \
-                                under parent directory of ino={parent} and name={parent_name:?}",
-                        ))?
-                    }
-                    SFlag::S_IFLNK => parent_node
-                        .load_child_symlink(child_name, child_attr)
-                        .await
-                        .context(format!(
-                            "lookup_helper() failed to read child symlink name={child_name:?} \
-                                under parent directory of ino={parent} and name={parent_name:?}",
-                        ))?,
-                    _ => panic!("lookup_helper() found unsupported file type={child_type:?}",),
-                };
-                (child_node, parent_name)
+        let result: &mut Option<FuseAttr> = &mut None;
+        let txn_result = retry_txn!(DEFAULT_TXN_RETRY_TIMES, {
+            let mut txn = self.kv_engine.new_meta_txn();
+            let pre_check_res = self.lookup_pre_check(&mut txn, parent, child_name).await;
+            let (ino, _, _) = match pre_check_res {
+                Ok((ino, child_type, child_attr)) => (ino, child_type, child_attr),
+                Err(e) => {
+                    debug!("lookup() failed to pre-check, the error is: {}", e);
+                    return Err(e);
+                }
             };
-
-            debug_assert_eq!(ino, child_node.get_ino());
-            child_node.set_ino(ino);
-            let child_ino = ino;
-            let attr = child_node.lookup_attr();
-            let full_path = child_node.full_path().to_owned();
-            {
-                let mut cache = self.cache.write().await;
-                cache.insert(child_ino, child_node);
+            // pre-check success, the child exists
+            // get the child node
+            let child_inum_key = KeyType::INum2Node(ino);
+            let get_result = txn.get(&child_inum_key).await;
+            let child_node = match get_result {
+                Ok(option_value) => {
+                    // If we get None here, it means the child node is deleted(inconsistent)
+                    // and we should retry
+                    let value = match option_value {
+                        Some(value) => value,
+                        None => {
+                            debug!(
+                                "lookup() found the child ino={} is deleted(inconsistent)",
+                                ino
+                            );
+                            continue;
```

Comment:
Seems these two continue should directly return error. The first points out the inconsistency between the metadata information of the parent node and the key-value (kv) of the child node. This type of inconsistency may require centralized statistical analysis and discussion. The second "continue" is usually an indication of an error in etcd. It should return the error directly.

Reply:
"I think you're right about the first point. However, for the second one, I believe it might be better to try again. Maybe this error can be resolved by retrying, especially for etcd.